### DISCUSSION TITLE
Type Hints for Globals

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [python/typing#1091] [Q&A] [answered]
The module this.py loads data from a function in module other.py. The data objects are loaded from a database and made global (I'm sure there is a better way to do this but hey ho). Placing a type hint inside the function or in the global statement cause the errors shown below. How are globals annotated?

this.py
```
import other

def get_data():
    other.read_data()
```
other.py
```
def read_data():
    global X, Y
    with some_db('db_file')) as db:
        X = db.read('x')
        Y = db.read('y'))
```
```
X: int = db.read('x')
SyntaxError: annotated name 'X' can't be global
```
```
global X: int
SyntaxError: invalid syntax
```

**Comment by srittau** [comment, marked as answer]
You can just define `X: int` in the global scope without an assignment:

```python
X: int

def read_data():
    global X
    X = db_read()
```

**Reply by dineshbvadhia** [reply]
This works. Thank-you.

Related to this, why is  

`X: int, Y:int`

not liked but
```
X: int
Y: int 
```
is liked?

**Reply by JelleZijlstra** [reply]
That's simply how the syntax was defined in PEP 526. I think your suggested syntax would have some ambiguity issues.
